projboard/tests/test_article.py

```diff
@@ -77,53 +39,30 @@ def test_edit_object(self, create_articles, create_subjects):
         article.edit(subject=unsaved_subject)
         assert article.subject_id == subject
 
-    @pytest.mark.django_db
-    def test_search_objects(self, create_articles, create_subjects, create_user):
-        user1 = create_user[1]
-        article_list = [i for i in create_articles]
-        subject1 = create_subjects[1]
-
-        # search by title or a part of it
-        assert article_list[1] in Article.search_by_title("Test")
-
-        # search by user
-        assert article_list[1] in Article.search_by_user(user1)
+    def test_search_by_object(self, articles, subjects, users):
+        # Assign articles for user
+        article_list = [i for i in articles]
+        for i in range(len(article_list)):
+            assert article_list[i] in Article.search_by_title("Test")
+            assert article_list[i] in Article.search_by_user(users[i])
+            assert article_list[i] in Article.search_by_subject(subjects[i])
 
-        # search by subject
-        assert article_list[1] in Article.search_by_subject(subject1)
-
-    @pytest.mark.django_db
-    def test_search_objects_none_exists(self, create_articles, create_subjects, create_user):
+    def test_search_non_exist_keys(self, articles, subjects, users):
         # if not in db
-        user1 = create_user[1]
-        article_list = [i for i in create_articles]
-        subject1 = create_subjects[1]
+        not_existed_user = users[DataType.NON_EXIST.value]
+        article_list = [i for i in articles]
+        not_existed_subject = subjects[DataType.NON_EXIST.value]
 
         # search by title or a part of it
         assert article_list[1] not in Article.search_by_title("Hakuna matata")
```

Comment:
change this also
article_list[DataType.EXIST.value]
And in line 60, 63 also
And 57,

It explains your logic

Reply:
@Yarboa Updated